v2/emailpassword/graphql-integration/backend-setup.mdx

```diff
@@ -32,69 +36,187 @@ Follow the [frontend and backend custom UI setup guides](../custom-ui/init/front
 
 ## b) Install GraphQL dependencies on your backend
 
-:::note
-We use express-graphql as an example, SuperTokens can be used with other GraphQL libraries.
+:::info
+We use Apollo server in this example, SuperTokens can be used with other GraphQL libraries as well.
 :::
 
 ```bash
-npm i express-graphql graphql
+npm install @apollo/server graphql
 ```
 
 ## c) Setup GraphQL
 
-### Create a schema 
+### Create your schema 
 
-```tsx
-import { buildSchema } from 'graphql';
+In this example we will create a simple social app that allows logged in users to create and view posts. To build an app like this you would need the following:
+- A way for users to login and logout
+- A way for users to create posts
+- A way for users to like posts
 
-var schema = buildSchema(`
-  type Query {
-    hello: String
-  }
-`);
+For sake of simplicity we will just focus on login/logout and viewing posts.
+
+From a GraphQL perspective we will need the following:
+- A query to fetch the user's posts
+
+```tsx
+const typeDefs = `
+type Query {
+    posts: [String],
+}
+`;
 ```
 
+The `posts` Query will return a list of `String` (one for each post) if the user is logged in.
+
+:::important
+We do not add any queries or mutations for login and logout because the SuperTokens middleware exposes APIs that handle this for us.
+:::
```

Comment:
to discuss: For graphql, for frameworks like hapi, how would the error handling work? I assume there too, apollo server would hijack the error and then our error handler (which is added app wide by us), would not work - to discuss tomorrow